### PR TITLE
Convert JavaNetAuthenticator to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp-urlconnection/src/main/java/okhttp3/JavaNetAuthenticator.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import java.net.Authenticator
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.net.Proxy
+
+/**
+ * Adapts [Authenticator] to [okhttp3.Authenticator]. Configure OkHttp to use [Authenticator] with
+ * [OkHttpClient.Builder.authenticator] or [OkHttpClient.Builder.proxyAuthenticator].
+ */
+@EverythingIsNonNull
```
Comment (JakeWharton) [inline okhttp-urlconnection/src/main/java/okhttp3/JavaNetAuthenticator.kt]: Delete?

File: okhttp-urlconnection/src/main/java/okhttp3/JavaNetAuthenticator.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    val url = request.url()
+    val proxyAuthorization = response.code() == 407
+    val proxy = route?.proxy() ?: Proxy.NO_PROXY
+
+    for (challenge in challenges) {
+      if (!"Basic".equals(challenge.scheme(), ignoreCase = true)) {
+        continue
+      }
+
+      val auth = when {
```
Comment (JakeWharton) [inline okhttp-urlconnection/src/main/java/okhttp3/JavaNetAuthenticator.kt]: Switch to
```kotlin
val auth = if (proxyAuthorization) {
  ..
} else {
  ..
}
```

File: okhttp-urlconnection/src/main/java/okhttp3/JavaNetAuthenticator.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp-urlconnection/src/main/java/okhttp3/JavaNetAuthenticator.kt]: Fixed!

File: okhttp-urlconnection/src/main/java/okhttp3/JavaNetAuthenticator.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp-urlconnection/src/main/java/okhttp3/JavaNetAuthenticator.kt]: Fixed.

**Pull request merged**